src/Renci.SshNet/Session.cs

```diff
@@ -1246,15 +1249,28 @@ private Message ReceiveMessage(Socket socket)
                     return null;
                 }
 
-                if (_serverCipher != null && !_serverAead && (_serverMac == null || !_serverEtm))
+                var plainFirstBlock = firstBlock;
+
+                // First block is not encrypted in AES GCM mode.
+                if (_serverCipher is not null
+#if NET6_0_OR_GREATER
+        and not Security.Cryptography.Ciphers.AesGcmCipher
+#endif
+                    )
                 {
-                    firstBlock = _serverCipher.Decrypt(firstBlock);
+                    _serverCipher.SetSequenceNumber(_inboundPacketSequence);
+
+                    // First block is not encrypted in ETM mode.
+                    if (_serverMac == null || !_serverEtm)
+                    {
+                        plainFirstBlock = _serverCipher.Decrypt(firstBlock);
+                    }
```

Comment:
This whole method makes me shed a tear. Maybe one day we can design something easier to comprehend. I have a bit of an idea

Reply:
It becomes "nobody knows why but it just works" 🤣 
Looking forward your idea.